File: model/message.go
package model

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Message struct {
	ID        primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	SessionID string             `bson:"session_id" json:"session_id"`
	UserName  string             `bson:"user_name" json:"username"`
	Content   string             `bson:"content" json:"content"`
	IsUser    bool               `bson:"is_user" json:"is_user"`
	CreatedAt time.Time          `bson:"created_at" json:"created_at"`
}

type History struct {
	IsUser  bool   `json:"is_user"`
	Content string `json:"content"`
}

File: node-backend/src/models/Message.js
import mongoose from 'mongoose';

const MessageSchema = new mongoose.Schema(
  {
    session_id: { type: String, required: true, index: true },
    user_name: { type: String, required: true, index: true },
    content: { type: String, required: true },
    is_user: { type: Boolean, required: true },
    citations: { type: Array, default: [] },
    rounds: { type: Number, default: null },
    created_at: { type: Date, default: Date.now, index: true },
  },
  { versionKey: false },
);

export default mongoose.models.Message || mongoose.model('Message', MessageSchema);
